cmd/mt-whisper-importer-reader/main.go

```diff
@@ -0,0 +1,328 @@
+		go processFromChan(fileChan, wg)
+	}
+
+	getFileListIntoChan(fileChan, wg)
+	wg.Wait()
+}
+
+func throwError(msg string) {
+	msg = fmt.Sprintf("%s\n", msg)
+	if *exitOnError {
+		panic(msg)
+	} else {
+		printLock.Lock()
+		fmt.Fprintln(os.Stderr, msg)
+		printLock.Unlock()
+	}
+}
+
+func log(msg string) {
+	if *verbose {
+		printLock.Lock()
+		fmt.Println(msg)
+		printLock.Unlock()
+	}
+}
+
+func processFromChan(files chan string, wg *sync.WaitGroup) {
+	client := &http.Client{}
+
+	for file := range files {
+		fd, err := os.Open(file)
+		if err != nil {
+			throwError(fmt.Sprintf("ERROR: Failed to open whisper file '%s': %s\n", file, err))
+			continue
+		}
+		w, err := whisper.OpenWhisper(fd)
+		if err != nil {
+			throwError(fmt.Sprintf("ERROR: Failed to open whisper file '%s': %s\n", file, err))
+			continue
+		}
+
+		log(fmt.Sprintf("Processing file %s", file))
+		b, err := getMetric(w, file).Encode()
+		if err != nil {
+			throwError(fmt.Sprintf("%s", err))
+		}
+
+		req, err := http.NewRequest("POST", *httpEndpoint, b)
+		if err != nil {
+			throwError(fmt.Sprintf("Cannot send request to http endpoint %s: %s", *httpEndpoint, err))
+		}
+		req.Header.Set("Content-Type", "application/json")
+		req.Header.Set("Content-Encoding", "gzip")
+
+		_, err = client.Do(req)
+		if err != nil {
+			throwError(fmt.Sprintf("Error sending request to http endpoint %s: %s", *httpEndpoint, err))
+		}
+
+		wg.Done()
+	}
+}
+
+// generate the metric name based on the file name and given prefix
+func getMetricName(file string) string {
+	for {
+		if file[:1] == "/" {
+			file = file[1:]
+		} else {
+			break
+		}
+	}
+	splits := strings.Split(file, "/")
+	leafNode := strings.Split(splits[len(splits)-1], ".")
+	splits[len(splits)-1] = leafNode[0]
+	return *namePrefix + strings.Join(splits, ".")
+}
+
+// pointSorter sorts points by timestamp
+type pointSorter []whisper.Point
+
+func (a pointSorter) Len() int           { return len(a) }
+func (a pointSorter) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a pointSorter) Less(i, j int) bool { return a[i].Timestamp < a[j].Timestamp }
+
+// the whisper archives are organized like a ringbuffer. since we need to
+// insert the points into the chunks in order we first need to sort them
+func sortPoints(points pointSorter) pointSorter {
+	sort.Sort(points)
+	return points
+}
+
+func getMetric(w *whisper.Whisper, file string) *archive.Metric {
+	if len(w.Header.Archives) == 0 {
+		throwError(fmt.Sprintf("ERROR: Whisper file contains no archives: %s", file))
+	}
+
+	archives := make([]archive.Archive, 0, len(w.Header.Archives))
+	var chunkSpan uint32
+	var base_id string
+	name := getMetricName(file)
+
+	for archiveIdx, archiveInfo := range w.Header.Archives {
+		md := getMetricData(name, int(archiveInfo.SecondsPerPoint))
+
+		if archiveIdx > 0 {
+			md.Id = api.AggMetricKey(
+				base_id,
+				w.Header.Metadata.AggregationMethod.String(),
+				archiveInfo.SecondsPerPoint,
+			)
+		} else {
+			base_id = md.Id
```

Comment:
this is a confusing and too complicated way to set base_id and m.Id. it relies on the order of archives, and uses a conditional clause that is not needed.  btw go uses camelcase so it would be baseId. set it after SetId is called (i.e. after getMetricData). Btw the `interval` in the MetricData and MetricDefinition structs refers to the raw interval. not the consolidated interval. since the interval is used in the metric id's, the way you have it now results in a completely different key which should not happen.